[JSC] Expose $addressBits to skip tests than need a lot of virtual address space

aoikonomopoulos opened the pull request:
#### e04108fd6a039c9d936017a0ce4cd7907d7ea4ed
<pre>
[JSC] Expose $addressBits to skip tests than need a lot of virtual address space
<a href="https://bugs.webkit.org/show_bug.cgi?id=255144">https://bugs.webkit.org/show_bug.cgi?id=255144</a>

Reviewed by Justin Michaud.

This achieves the same effect, while making the skip self-explanatory.
Also, the list of architectures that need or don&apos;t need to skip a test
because of VA limitations can be managed in one place.

* JSTests/stress/big-wasm-memory-grow-no-max.js:
* JSTests/stress/big-wasm-memory-grow.js:
* JSTests/stress/big-wasm-memory.js:
* JSTests/stress/typed-array-always-large.js:
* JSTests/stress/typed-array-eventually-large.js:
* JSTests/stress/typed-array-large-eventually-oob.js:
* JSTests/stress/typed-array-large-slice.js:
* JSTests/wasm/regress/242294.js:
* JSTests/wasm/regress/llint-callee-saves-with-fast-memory.js:
* JSTests/wasm/stress/externref-result-tuple.js:
* JSTests/wasm/stress/only-referenced.js:
* JSTests/wasm/v8/huge-memory.js:
* JSTests/wasm/v8/huge-typedarray.js:
* Tools/Scripts/run-jsc-stress-tests:

Canonical link: <a href="https://commits.webkit.org/262824@main">https://commits.webkit.org/262824@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d750dabe510d1913c923612d43fd50f5d105efe3

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/2699 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/2700 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/2841 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/4107 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/3069 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/2663 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/2843 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/2797 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/2376 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/2724 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/3166 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/2412 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/3879 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/660 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/2397 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/2263 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/2261 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/2782 "Built successfully") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/2416 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/3632 "260 api tests failed or timed out") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/2580 "Built successfully and passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/2448 "Passed tests") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/2230 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/2776 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/2451 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/2400 "Passed tests") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/655 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/671 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/2442 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/2837 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/2595 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/770 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

justinmichaud commented on Tools/Scripts/run-jsc-stress-tests:
```diff
@@ -589,6 +589,18 @@ $isFTLPlatform = !($architecture == "x86" || $architecture == "arm" || $architec
+# don't need to be this precise.
+$addressBits = case $architecture
+               when /.*64.*/
+                   48
```
nit: iOS is 36-bit. I'm not sure it matters here

aoikonomopoulos replied:
Did not know that. Changed the line to 36 instead -- we can worry about differentiating between 64-bit iOS and non-iOS when we get tests that require > 36 bits of address space.

justinmichaud reviewed (approved):
r=me once EWS passes

aoikonomopoulos committed:
[JSC] Expose $addressBits to skip tests than need a lot of virtual address space
https://bugs.webkit.org/show_bug.cgi?id=255144

Reviewed by Justin Michaud.

This achieves the same effect, while making the skip self-explanatory.
Also, the list of architectures that need or don't need to skip a test
because of VA limitations can be managed in one place.

* JSTests/stress/big-wasm-memory-grow-no-max.js:
* JSTests/stress/big-wasm-memory-grow.js:
* JSTests/stress/big-wasm-memory.js:
* JSTests/stress/typed-array-always-large.js:
* JSTests/stress/typed-array-eventually-large.js:
* JSTests/stress/typed-array-large-eventually-oob.js:
* JSTests/stress/typed-array-large-slice.js:
* JSTests/wasm/regress/242294.js:
* JSTests/wasm/regress/llint-callee-saves-with-fast-memory.js:
* JSTests/wasm/stress/externref-result-tuple.js:
* JSTests/wasm/stress/only-referenced.js:
* JSTests/wasm/v8/huge-memory.js:
* JSTests/wasm/v8/huge-typedarray.js:
* Tools/Scripts/run-jsc-stress-tests:

Canonical link: https://commits.webkit.org/262824@main
